data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-transform/src/test/java/org/eclipse/edc/protocol/dsp/transferprocess/transformer/to/JsonObjectToTransferStartMessageTransformerTest.java

```diff
@@ -15,44 +15,104 @@
 package org.eclipse.edc.protocol.dsp.transferprocess.transformer.to;
 
 import jakarta.json.Json;
+import jakarta.json.JsonObject;
 import org.eclipse.edc.protocol.dsp.spi.types.HttpMessageProtocol;
 import org.eclipse.edc.protocol.dsp.transferprocess.transformer.type.to.JsonObjectToTransferStartMessageTransformer;
+import org.eclipse.edc.spi.types.domain.DataAddress;
 import org.eclipse.edc.transform.spi.TransformerContext;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.CONTEXT;
 import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspTransferProcessPropertyAndTypeNames.DSPACE_DATAADDRESS_TYPE;
 import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspTransferProcessPropertyAndTypeNames.DSPACE_PROCESSID_TYPE;
 import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspTransferProcessPropertyAndTypeNames.DSPACE_SCHEMA;
 import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspTransferProcessPropertyAndTypeNames.DSPACE_TRANSFER_START_TYPE;
 import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 class JsonObjectToTransferStartMessageTransformerTest {
 
     private final String processId = "TestProcessId";
 
+    private final String type = "AWS";
+
+    private final String propertyKey = "region";
+
+    private final String propertyValue = "europe";
+
     private TransformerContext context = mock(TransformerContext.class);
 
     private JsonObjectToTransferStartMessageTransformer transformer;
 
     @BeforeEach
     void setUp() {
+        var dataAddress = DataAddress.Builder.newInstance()
+                .type(type)
+                .property(propertyKey, propertyValue)
+                .build();
         transformer = new JsonObjectToTransferStartMessageTransformer();
+
+        when(context.transform(isA(JsonObject.class), eq(DataAddress.class))).thenReturn(dataAddress);
```

Comment:
there's no need to put this in the setup, as it would be necessary only for test number 2, so please put it there, no reason to have the `type`, `propertyKey` and `propertyFields` defined as well

Reply:
You need a type, because DataAddress needs a type or other fields to be filled out. So i chosed to add an type field, to make it easy. I can delete the property fields, to make it look less (easier).